common/pkg/capi/clustertopology/handlers/mutation/meta.go

```diff
@@ -59,6 +68,18 @@ func (mgp metaGeneratePatches) GeneratePatches(
 	resp *runtimehooksv1.GeneratePatchesResponse,
 ) {
 	clusterKey := handlers.ClusterKeyFromReq(req)
+	getCluster := func(ctx context.Context) (*clusterv1.Cluster, error) {
+		cluster := &clusterv1.Cluster{}
+		var err error
+		fn := sync.OnceFunc(func() {
+			err = mgp.cl.Get(ctx, clusterKey, cluster)
+		})
+		fn()
```

Comment:
Why is `sync.OnceFunc` needed?

Reply:
This would be so it would happen once per cluster mutation.